test/framework/self-tests/fail_assert.c

```diff
@@ -0,0 +1,25 @@
+/**
+ * @file test/self-tests/fail_assert.c
+ *
+ * @brief Test: Test core functions of the testing framework
+ *
+ * This is the test that checks if assert can fail a test.
+ *
+ * SPDX-FileCopyrightText: 2008-2021 HPDCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+#include <test.h>
+#include <string.h>
+
+extern test_ret_t test_pass(__unused void *_);
+extern test_ret_t test_pass_assert(__unused void *_);
+
+int main(void)
+{
+	init(0);
+
+	test_xf("Test unexpected pass", test_pass, NULL);
+	test_xf("Test unexpected pass with asserts", test_pass_assert, NULL);
```

Comment:
This should be another separate test case, else one of the two ```test_xf()``` call could incorrectly pass and the test would still correctly fail.

Reply:
Good one!